python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -1141,6 +1156,27 @@ def submit_run_request(
                     )
                     updated_evaluation_keys.add(entity_key)
 
+            now = get_current_timestamp()
+            if now - last_status_poll_time >= poll_interval:
```

Comment:
An alternate implementation of this is to check the sensor status after every N run submissions. I went with the time approach so that it is easier to understand how quickly the sensor will respond after you turn it off

Reply:
updated this to poll after a number of run are submitted. Another alt implementation would be to make the submission fn take a batch of run requests, and then we check after each thread completes rather than keeping track of a counter, but idk if there is a reason to have each thread only submit one run and the counter version seems fine?